hyperloglog.go

```diff
@@ -71,21 +72,12 @@ func (h *HyperLogLog) Reset() {
 	}
 }
 
-// Calculate the position of the leftmost 1-bit.
-func rho(val uint32, max uint32) uint8 {
-	r := uint32(1)
-	for val&0x80000000 == 0 && r <= max {
-		r++
-		val <<= 1
-	}
-	return uint8(r)
-}
-
 // Add to the count. val should be a 32 bit unsigned integer from a
 // good hash function.
 func (h *HyperLogLog) Add(val uint32) {
 	k := 32 - h.B
-	r := rho(val<<h.B, k)
+	slice := val<<h.B | 1<<(h.B-1)
```

Comment:
nit: maybe add a panic in `func New(registers)` if registers == 0 so `h.B` is guaranteed to be > 0?

Reply:
And I would put parens to make precedence explicit: `(val<<h.B) | (1<<(h.B-1))`

Reply:
^ agreed re: precedence, but interestingly logical AND and OR [come last per the go specification](https://go.dev/ref/spec#Operator_precedence) though